Projects/Shapes/Shapes/Shapes/StaticArray.h

```diff
@@ -0,0 +1,92 @@
+#pragma once
+
+template<typename T, int N>
+class TStaticArray
+{
+private:
+	int Size = N;
+	T Array[N] = {};
+
+public:
+
+	//-------------------------------------------------------------------------------------------
+	T& operator[](const int Index)
+	{
+		return Array[Index];
+	};
+
+	//-------------------------------------------------------------------------------------------
+	const T& operator[](const int Index) const 
+	{
+		return Array[Index];
+	};
+
+	//-------------------------------------------------------------------------------------------
+	T& Front()
+	{
+		return Array[0];
+	};
+
+	//-------------------------------------------------------------------------------------------
+	const T& Front() const
+	{
+		return Array[0];
+	}
+
+	//-------------------------------------------------------------------------------------------
+	T& Back()
+	{
+		return Array[Size - 1];
+	};
+
+	//-------------------------------------------------------------------------------------------
+	const T& Back() const
+	{
+		return Array[Size - 1];
+	};
+
+	//-------------------------------------------------------------------------------------------
+	T* GetData()
+	{
+		return Array;
+	};
+
+	//-------------------------------------------------------------------------------------------
+	const T* GetData() const
+	{
+		return Array;
+	};
+
+	//-------------------------------------------------------------------------------------------
+	bool IsEmpty() const
+	{
+		return Size == 0;
+	};
+
+	//-------------------------------------------------------------------------------------------
+	int GetSize() const
+	{
+		return Size;
+	};
+
+	//-------------------------------------------------------------------------------------------
+	void Fill(T& Filler)
```

Comment:
Is this Filler changing?
![image](https://user-images.githubusercontent.com/36243539/191055254-13433a3d-8c8c-400e-a45d-e440743fc494.png)

Reply:
![image](https://user-images.githubusercontent.com/44332093/191064879-cc052f16-9add-4ef6-a7d6-0288542fa971.png)

In other news I'm pretty sure I improved on the const parameters for the next project. On the primitive data types I remember to place const but somehow for the ones that matter the most I forget. I think it's related to the color coding (const and primitive data types are same color x3 )